lighthouse-core/audits/dobetterweb/has-inspector-issues.js

```diff
@@ -0,0 +1,246 @@
+   * @return {Array<IssueItem>}
+   */
+  static getSameSiteCookieItems(sameSiteCookieIssues) {
+    if (!sameSiteCookieIssues) {
+      return [];
+    }
+
+    return sameSiteCookieIssues.map(issue => {
+      const requestUrl = issue.request && issue.request.url;
+      return {
+        issueType: 'SameSiteCookie',
+        description: str_(UIStrings.sameSiteMessage),
+        requestUrl: issue.cookieUrl || requestUrl,
+      };
+    });
+  }
+
+  /**
+   * @param {Array<LH.Crdp.Audits.BlockedByResponseIssueDetails>} blockedByResponseIssues
+   * @return {Array<IssueItem>}
+   */
+  static getBlockedByResponseItems(blockedByResponseIssues) {
+    if (!blockedByResponseIssues) {
+      return [];
+    }
+
+    return blockedByResponseIssues.map(issue => {
+      const blockedReason = issue.reason;
+      return {
+        issueType: 'BlockedByResponse',
+        description: blockedByResponseMsgMap[blockedReason],
+        requestUrl: issue.request.url,
+      };
+    });
+  }
+
+  /**
+   * @param {Array<LH.Crdp.Audits.HeavyAdIssueDetails>} heavyAdsIssues
+   * @return {Array<IssueItem>}
+   */
+  static getHeavyAdsItems(heavyAdsIssues) {
+    if (!heavyAdsIssues) {
+      return [];
+    }
+
+    return heavyAdsIssues.map(issue => {
+      const reason = issue.reason;
+      return {
+        issueType: 'HeavyAds',
+        description: heavyAdsMsgMap[reason],
+      };
+    });
+  }
+
+  /**
+   * @param {Array<LH.Crdp.Audits.ContentSecurityPolicyIssueDetails>} cspIssues
+   * @return {Array<IssueItem>}
+   */
+  static getContentSecurityPolicyItems(cspIssues) {
+    if (!cspIssues) {
+      return [];
+    }
+
+    return cspIssues
+    .filter(issue => {
+      // kTrustedTypesSinkViolation and kTrustedTypesPolicyViolation aren't currently supported by the Issues panel
+      return issue.contentSecurityPolicyViolationType !== 'kTrustedTypesSinkViolation' &&
+        issue.contentSecurityPolicyViolationType !== 'kTrustedTypesPolicyViolation';
+    })
+    .map(issue => {
+      const blockedUrl = issue.blockedURL;
+      return {
+        issueType: 'ContentSecurityPolicy',
+        description: contentSecurityPolicyMsgMap[issue.contentSecurityPolicyViolationType],
+        requestUrl: blockedUrl,
+      };
+    });
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit(artifacts) {
+    /** @type {LH.Audit.Details.Table['headings']} */
+    const headings = [
+      {key: 'issueType', itemType: 'text', text: str_(UIStrings.columnIssueType)},
+      {key: 'description', itemType: 'text', text: str_(i18n.UIStrings.columnDescription)},
+      {key: 'requestUrl', itemType: 'url', text: 'Request URL'},
+    ];
+
+    const issues = artifacts.InspectorIssues;
+    /** @type {Array<IssueItem>} */
+    const items = [];
+
+    for (const [issueType, issuesOfType] of Object.entries(issues)) {
+      switch (issueType) {
```

Comment:
`switch` is kind of overkill here, since we know all the things that we want pushed and we know that they're all defined (as least as an empty array). Something like

```js
const issues = artifacts.InspectorIssues;
const items = [
  ...this.getSameSiteCookieItems(issues.sameSiteCookies),
  ...this.getMixedContentItems(issues.mixedContent),
  ...this.getBlockedByResponseItems(issues.blockedByResponse),
  ...this.getHeavyAdsItems(issues.heavyAds),
  ...this.getContentSecurityPolicyItems(issues.contentSecurityPolicy),
];
```